packages/gatsby/src/query/query-compiler.js

```diff
@@ -303,31 +310,9 @@ const processDefinitions = ({
     .map(([name, _]) => name)
 
   for (const operation of operations) {
-    const name = operation.name.value
+    const name = operation.def.name.value
     const originalDefinition = definitionsByName.get(name)
-    const filePath = definitionsByName.get(name).filePath
-
-    // Check for duplicate page/static queries in the same component.
-    // (config query is not a duplicate of page/static query in the component)
-    // TODO: make sure there is at most one query type per component (e.g. one config + one page)
-    if (processedQueries.has(filePath) && !originalDefinition.isConfigQuery) {
-      const otherQuery = processedQueries.get(filePath)
-
-      addError(
-        multipleRootQueriesError(
-          filePath,
-          originalDefinition.def,
-          otherQuery && definitionsByName.get(otherQuery.name).def
-        )
-      )
-
-      store.dispatch(
-        actions.queryExtractionGraphQLError({
-          componentPath: filePath,
-        })
-      )
-      continue
-    }
+    const { filePath, templatePath } = operation
```

Comment:
Move this destructuring to the top of this scope, also destructure `def` and then you can replace `operation.def` with `def`